From 0a15889fb9057ea6afe4e311fce33dd679587029 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Fri, 9 Jan 2015 11:54:48 -0800
Subject: [PATCH 2/8] keymaster: Set HEAP_MASK_COMPATIBILITY by platform for
 QCOM_HARDWARE

* Google kernels have a separate hack to handle this, only set by
  platform for QCOM_HARDWARE devices.

Change-Id: Iab91cee4922e1b3467691a04eb1c5061705fbab3
---
 Android.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Android.mk b/Android.mk
index a078c8c..39fe7da 100644
--- a/Android.mk
+++ b/Android.mk
@@ -5,9 +5,15 @@ LOCAL_PATH := $(call my-dir)
 ifneq ($(filter msm8960 msm8226 msm8974 msm8610 msm8084 apq8084 msm8909 msm8916 msm8952 msm8992 msm8994,$(TARGET_BOARD_PLATFORM)),)
 
 keymaster-def := -fvisibility=hidden -Wall
+ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
+ifneq ($(filter msm8960 msm8226 msm8974 msm8610 msm8084 apq8084,$(TARGET_BOARD_PLATFORM)),)
+keymaster-def += -D_ION_HEAP_MASK_COMPATIBILITY_WA
+endif
+else
 ifeq ($(TARGET_BOARD_PLATFORM),msm8084)
 keymaster-def += -D_ION_HEAP_MASK_COMPATIBILITY_WA
 endif
+endif
 
 include $(CLEAR_VARS)
 
-- 
2.17.1

